reports/models.py

```diff
@@ -0,0 +1,459 @@
+
+    def unsynced(self):
+        """
+        Query unsynced enrolment report instances:
+        Enrolment report is not in sync when it's related enrolment instance
+        has an updated_at -field value greater than
+        report instances updated_at -field value.
+        """
+        # Even if the enrolment instance has updated,
+        # we do not want to update it's related enrolment report instance,
+        # until the status of enrolment has changed.
+        enrolment_ids_with_statuses = list(
+            {"_enrolment_id": id, "enrolment_status": status}
+            for id, status in EnrolmentQuerySet.unsynced(self).values_list(
+                "id", "status"
+            )
+        )
+        separator = ":"
+        return (
+            # Filter all reports that are related to enrolments
+            self.filter(
+                _enrolment_id__in=[
+                    entry["_enrolment_id"] for entry in enrolment_ids_with_statuses
+                ]
+            )
+            # Exclude from the queryset all the report instances
+            # that are in the enrolment_ids_with_statuses list
+            # by using the annotated concat-field
+            .annotate(
+                # Annotate a new field that concatenates enrolment id and status
+                enrolment_id_status=models.functions.Concat(
+                    "_enrolment_id",
+                    models.Value(separator),
+                    "enrolment_status",
+                    output_field=models.CharField(),
+                ),
+            ).exclude(
+                enrolment_id_status__in=[
+                    "{}{}{}".format(
+                        entry["_enrolment_id"], separator, entry["enrolment_status"]
+                    )
+                    for entry in enrolment_ids_with_statuses
+                ]
+            )
+        )
+
+
+class EnrolmentReportManager(models.Manager):
+    def get_queryset(self):
+        return UnsyncedQuerySet(self.model, using=self._db)
+
+    def latest_sync(self):
+        """Returns latest updated_at, which represents the date of the last sync made"""
+        return self.get_queryset().latest_sync()
+
+    def unsynced(self, sync_from: datetime = None):
+        """Query unsynced enrolment report instances"""
+        return self.get_queryset().unsynced()
+
+    def has_missing(self) -> bool:
+        # TODO: Find n effective and more reliable query to check this.
+        return bool(self.count_missing())
+
+    def count_missing(self) -> int:
+        return (
+            occurrences_models.Enrolment.objects.all().count()
+            - self.model.objects.all().exclude(_enrolment__isnull=True).count()
+        )
+
+    def create_missing(
+        self, hydrate_linkedevents_event=False, sync_from: datetime = None
+    ):
+        """Create missing enrolment report instances."""
+        enrolments = self.model.enrolment_objects.missing(sync_from=sync_from)
```

Comment:
If `EnrolmentQuerySet` is turned into functions this could be:

```suggestion
        enrolments = get_missing_enrollments(sync_from=sync_from)
```